app/controllers/IncomeTaxComparisonController.scala

```diff
@@ -118,7 +116,8 @@ class IncomeTaxComparisonController @Inject()(
 
           Ok(mainView(model, applicationConfig))
         }
-        case _ => throw new RuntimeException("Not able to fetch income tax comparision details")
+        case _ =>
```

Comment:
Should we return a 500 if any of the services returns a TaiFailureResponse? 

Should we gather more information about why this couldn't happen? Which were the culprits to error etc? 
Have we considered alternatives for the API rather than returning Future[TaiResponse], return Future[Either[Error, Payload]] or Future[Payload] etc? 
Also I think